community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/DiskLayer.java

```
@@ -707,68 +812,273 @@ public String propertyKeyGetName( int propertyKeyId )
+    public Iterator<DefinedProperty> relationshipGetAllProperties( StoreStatement statement, final long relationshipId )
             throws EntityNotFoundException
     {
-        return propertyLoader.relLoadProperties( relationshipId, new IteratingPropertyReceiver() );
-//        catch ( InvalidRecordException e )
-//        {
-//            throw new EntityNotFoundException( EntityType.RELATIONSHIP, relationshipId, e );
-//        }
+        try ( StoreSingleRelationshipCursor storeRelationshipCursor = statement.acquireSingleRelationshipCursor() )
+        {
+            if ( !storeRelationshipCursor.init( relationshipId ).next() )
+            {
+                throw new EntityNotFoundException( EntityType.RELATIONSHIP, relationshipId );
+            }
+
+            final PropertyCursor cursor = storeRelationshipCursor.get().properties();
+
+            return new Iterator<DefinedProperty>()
+            {
+                boolean hasNext = cursor.next();
+
+                @Override
+                public boolean hasNext()
+                {
+                    return hasNext;
+                }
+
+                @Override
+                public DefinedProperty next()
+                {
+                    if ( hasNext )
+                    {
+                        try
+                        {
+                            return cursor.get().getProperty();
+                        }
+                        finally
+                        {
+                            hasNext = cursor.next();
+
+                            if ( !hasNext )
+                            {
+                                cursor.close();
+                            }
+                        }
+                    }
+                    else
+                    {
+                        throw new NoSuchElementException();
+                    }
+                }
+
+                @Override
+                public void remove()
+                {
+
```

you should throw `UnsupportedOperation()` here